Products/ZenModel/tests/testDevice.py

```diff
@@ -67,6 +67,32 @@ def testManage_createDeviceDupIp(self):
         self.assertRaises(DeviceExistsError,
           manage_createDevice, self.dmd, 'mydevice2', '/', manageIp='1.1.1.1')
 
+    def testManage_createDeviceWithIpFromInterface(self):
+        # create device with ip that is on Interface of another device
+        testIp = '1.2.3.4'
+
+        # Need a network interface on that device
+        from Products.ZenModel.IpInterface import IpInterface
+        tmpIface = IpInterface('testNIC')
+        self.dev.os.interfaces._setObject('testNIC', tmpIface)
+        self.iface1 = self.dev.getDeviceComponents()[0]
+        self.iface1.addIpAddress(testIp)
+
+        ip = self.dev.getNetworkRoot().findIp(testIp)
+        self.assert_(ip is not None)
+
+        dev = manage_createDevice(self.dmd, 'mydevice', '/', manageIp=testIp)
+        self.assert_(dev is not None)
```

Comment:
There should be a check to validate that the `self.dev` and `dev` devices have different `manageIp` values.